server/app/controllers/applicant/UpsellController.java

```diff
@@ -93,12 +100,13 @@ public CompletionStage<Result> considerRegister(
             .getReadOnlyApplicantProgramService(applicantId, programId)
             .toCompletableFuture();
 
-    return CompletableFuture.allOf(isCommonIntake, account, roApplicantProgramService)
+    return CompletableFuture.allOf(
+            isCommonIntake, account, roApplicantProgramService, isProgramDisabled)
         .thenComposeAsync(
             ignored -> {
-              if (!isCommonIntake.join()) {
-                // If this isn't the common intake form, we don't need to make the
-                // call to get the applicant's eligible programs.
+              if (!isCommonIntake.join() || isProgramDisabled.join()) {
```

Comment:
**question:** What is a scenario that the user would get here in a disabled program? I thought that this was the page that shows after a user finishes filling out an application?